src/serveressFunction/build-run-deploy.ts

```diff
@@ -0,0 +1,219 @@
+            }
+            return null;
+        } catch (err) {
+            return null;
+        }
+    }
+
+    public async buildFunction(functionName: string, functionUri: Uri): Promise<void> {
+        const exisitingTerminal = this.buildTerminalMap.get(`build-${functionUri.fsPath}`);
+        const outputEmitter = this.buildEmiterMap.get(`build-${functionUri.fsPath}`);
+        if (exisitingTerminal) {
+            let exisitingProcess = this.buildPrcessMap.get(exisitingTerminal);
+            void window.showWarningMessage(`Do you want to restart ${functionName} build ?`, 'Yes', 'No').then(async (value: string) => {
+                if (value === 'Yes') {
+                    exisitingTerminal.show(true);
+                    await commands.executeCommand('workbench.action.terminal.clear')
+                    outputEmitter.fire(`Start Building ${functionName} \r\n`);
+                    exisitingProcess.kill('SIGINT')
+                    this.buildPrcessMap.delete(exisitingTerminal);
+                    const clusterVersion: ClusterVersion | null = await this.checkOpenShiftCluster();
+                    const buildImage = await this.getImage(functionUri);
+                    const opt: SpawnOptions = { cwd: functionUri.fsPath };
+                    void CliChannel.getInstance().spawnTool(ServerlessCommand.buildFunction(functionUri.fsPath, buildImage, clusterVersion), opt).then((cp) => {
+                        exisitingProcess = cp;
+                        this.buildPrcessMap.set(exisitingTerminal, cp);
+                        exisitingProcess.on('error', (err) => {
+                            void window.showErrorMessage(err.message);
+                        });
+                        exisitingProcess.stdout.on('data', (chunk) => {
+                            outputEmitter.fire(`${chunk as string}`.replaceAll('\n', '\r\n'));
+                        });
+                        exisitingProcess.stderr.on('data', (errChunk) => {
+                            outputEmitter.fire(`\x1b[31m${errChunk as string}\x1b[0m`.replaceAll('\n', '\r\n'));
+                            void window.showErrorMessage(`${errChunk as string}`);
+                        });
+                        exisitingProcess.on('exit', () => {
+                            outputEmitter.fire('\r\nPress any key to close this terminal\r\n');
+                        });
+                    });
+                }
+            });
+        } else {
+            await this.buildProcess(functionUri, functionName);
+        }
+    }
+
+    private async buildProcess(functionUri: Uri, functionName: string) {
+        const clusterVersion: ClusterVersion | null = await this.checkOpenShiftCluster();
+        const buildImage = await this.getImage(functionUri);
+        const outputEmitter = new EventEmitter<string>();
+        let devProcess: ChildProcess;
+        let terminal = window.createTerminal({
```

Comment:
See `./src/webview/openshift-terminal/openShiftTerminal.ts`, `OpenShiftTerminalManager.createTerminal()` in order to integrate into the separate OpenShift Terminal. If there is something missing in the API, please let me know so that I can fix it.

Reply:
The integration currently blocked by https://github.com/redhat-developer/vscode-openshift-tools/issues/3031

Reply:
Closing this comment. See https://github.com/redhat-developer/vscode-openshift-tools/pull/3041, which reverts the OpenShift Terminal